---
title: "thesis"
author: "Shriram"
date: "2/8/2022"
output: html_document
---


#the libraries
```{r}
library("tidyverse")
library("plotly")
library("DT")
library("ggbeeswarm")
library("knitr")
library("ggrepel")
library("genetics")
library("ggnewscale")
library("cowplot")
library("gggenes")
```


#Population distribution : box plots

Box plots of c.e. telomere lengths:

```{r}
ce_len <- read_tsv('/projects/b1059/projects/Shriram/data/elegans/lengths/eleg_strain.tsv') %>% arrange(desc(length))
ce_len_isotype <- read_tsv('/projects/b1059/projects/Shriram/data/elegans/lengths/eleg_mean.tsv')


pe <- ce_len %>% mutate(strain=fct_reorder(strain,length)) %>%
      ggplot(aes(x=strain,y=length)) + 
        xlab("elegans strains")+ 
        geom_bar(stat='identity',fill="#f68060") 
pe
```

Box plots of c.b telomere lengths: 

```{r}
cb_len <- as_tibble(read_tsv('/projects/b1059/projects/Shriram/data/briggsae/lengths/brig_strain.tsv')) %>% arrange(desc(length))
cb_len_isotype <- read_tsv('/projects/b1059/projects/Shriram/data/briggsae/lengths/brig_mean.tsv')


pb <- cb_len %>% mutate(strain=fct_reorder(strain,length)) %>%
      ggplot(aes(x=strain,y=length)) + 
        geom_bar(stat='identity',fill="#f68060") +
        xlab("briggsae strains")

pb
```

Box plots of c.t telomere lengths: 

```{r}
ct_len <- as_tibble(read_tsv('/projects/b1059/projects/Shriram/data/tropicalis/lengths/trop_strain.tsv')) %>% arrange(desc(length))
ct_len_isotype <- read_tsv('/projects/b1059/projects/Shriram/data/tropicalis/lengths/trop_mean.tsv')


pt <- ct_len %>% mutate(strain=fct_reorder(strain,length)) %>%
      ggplot(aes(x=strain,y=length)) + 
        xlab("tropicalis strains")+ 
        geom_bar(stat='identity',fill="#f68060")
pt
```
#Manhattan plots:

1. c.e. Manhattan plot

```{r}

# in nextflow use sed to edit this field and make a copy of the .rmd for each trait
trait_name <- "telomere length"

#load gene list file for labelling
eleg_genes <- read.delim("/projects/b1059/projects/Shriram/data/elegans/goi/eleg_goi.tsv")

# load independent tests result
total_independent_tests <- read.table("/projects/b1059/projects/Shriram/data/elegans/nemascan_runs/Analysis_Results-20211219/Genotype_Matrix/total_independent_tests.txt", quote="\"", comment.char="", stringsAsFactors=FALSE)

independent_test_cutoff <- -log10(0.05/total_independent_tests[[1]])

# load processed mapping data. 
#processed_mapping <- read.delim("/projects/b1059/projects/Shriram/worm/results/result_date/10_12/Analysis_Results-20211012/Mapping/Processed/processed_telomere_length_AGGREGATE_mapping.tsv", stringsAsFactors=FALSE) %>%
processed_mapping <- read.delim("/projects/b1059/projects/Shriram/data/elegans/nemascan_runs/Analysis_Results-20211219/Mapping/Processed/processed_length_AGGREGATE_mapping.tsv", stringsAsFactors=FALSE) %>%
  dplyr::mutate(CHROM = factor(CHROM, levels = c("I","II","III","IV","V","X","MtDNA"))) %>%
  dplyr::select(-marker) %>%
  tidyr::unite("marker", CHROM, POS, sep = ":", remove = F)


for.plot <- processed_mapping %>%
  dplyr::mutate(CHROM = as.factor(CHROM)) %>%
  dplyr::filter(CHROM != "MtDNA") %>%
  dplyr::mutate(algorithm = as.factor(algorithm))

BF <- processed_mapping %>% 
  dplyr::group_by(trait) %>% 
  dplyr::filter(log10p != 0) %>% 
  dplyr::mutate(BF = -log10(0.05/sum(log10p > 0, na.rm = T))) %>%
  dplyr::ungroup() %>%
  dplyr::select(BF) %>%
  unique(.) %>%
  as.numeric()


# ntests <- data.table::fread(tests) %>%
#  as.numeric()
# EIGEN <- -log10(0.05/ntests)
BF.frame <- processed_mapping %>%
  dplyr::select(trait) %>%
  dplyr::filter(!duplicated(trait)) %>%
  dplyr::mutate(BF = BF, EIGEN  = independent_test_cutoff)

for.plot.ann <- for.plot %>%
  dplyr::mutate(sig = case_when(log10p > BF.frame$BF ~ "BF",
                                log10p > BF.frame$EIGEN ~ "EIGEN",
                                TRUE ~ "NONSIG"))

sig.colors <- c("red","#EE4266")
names(sig.colors) <- c("BF","EIGEN")

man.plot <-  ggplot2::ggplot() + 
  ggplot2::theme_bw() + 
  ggplot2::geom_point(data = for.plot.ann[which(for.plot.ann$sig != "NONSIG"),], 
                      mapping = aes(x = POS/1000000, 
                                    y = log10p,
                                    colour = sig),
                                    size=0.25)+
  ggplot2::scale_colour_manual(values = sig.colors) + 
  ggplot2::geom_point(data = for.plot[which(for.plot.ann$sig == "NONSIG"),], 
                      mapping = aes(x = POS/1000000, 
                                   y = log10p), 
                                   alpha = 0.5,
                                   size=0.25) +
  ggplot2::scale_y_continuous(expand = c(0,0), limits = c(0,BF + 6.8)) +
  scale_x_continuous(expand = c(0, 0), breaks = c(5, 10, 15, 20)) +
  ggplot2::geom_point(data = eleg_genes, 
                      mapping=aes(x = POS/1000000, 
                                  y = log10p
                                  ),
                                  shape=25,
                                  size=2,
                                  fill="blue") +
  ggplot2::geom_hline(data = BF.frame, aes(yintercept = BF), linetype = 2) + 
  ggplot2::geom_hline(data = BF.frame, aes(yintercept = EIGEN), linetype = 3) + 
  ggplot2::labs(x = "",
                y = "" ) +
  ggplot2::theme(legend.position = "none", 
                 panel.grid = element_blank()) + 
  ggplot2::facet_grid(. ~ CHROM, scales = "free_x", space = "free") 

eleg_manplot <- man.plot
eleg_manplot
```

2. c.b. Manhattan plot

```{r}
# in nextflow use sed to edit this field and make a copy of the .rmd for each trait
trait_name <- "telomere length"

#load gene list file for labelling
brig_genes <- read.delim("/projects/b1059/projects/Shriram/data/briggsae/goi/brig_goi.tsv")

# load independent tests result
total_independent_tests <- read.table("/projects/b1059/projects/Shriram/data/briggsae/nemascan_runs/Analysis_Results-20211219/Genotype_Matrix/total_independent_tests.txt", quote="\"", comment.char="", stringsAsFactors=FALSE)

independent_test_cutoff <- -log10(0.05/total_independent_tests[[1]])

# load processed mapping data. 
processed_mapping <- read.delim("/projects/b1059/projects/Shriram/data/briggsae/nemascan_runs/Analysis_Results-20211219/Mapping/Processed/processed_length_AGGREGATE_mapping.tsv", stringsAsFactors=FALSE) %>%
  dplyr::mutate(CHROM = factor(CHROM, levels = c("I","II","III","IV","V","X","MtDNA"))) %>%
  dplyr::select(-marker) %>%
  tidyr::unite("marker", CHROM, POS, sep = ":", remove = F)


for.plot <- processed_mapping %>%
  dplyr::mutate(CHROM = as.factor(CHROM)) %>%
  dplyr::filter(CHROM != "MtDNA") %>%
  dplyr::mutate(algorithm = as.factor(algorithm))

BF <- processed_mapping %>% 
  dplyr::group_by(trait) %>% 
  dplyr::filter(log10p != 0) %>% 
  dplyr::mutate(BF = -log10(0.05/sum(log10p > 0, na.rm = T))) %>%
  dplyr::ungroup() %>%
  dplyr::select(BF) %>%
  unique(.) %>%
  as.numeric()


# ntests <- data.table::fread(tests) %>%
#  as.numeric()
# EIGEN <- -log10(0.05/ntests)
BF.frame <- processed_mapping %>%
  dplyr::select(trait) %>%
  dplyr::filter(!duplicated(trait)) %>%
  dplyr::mutate(BF = BF, EIGEN  = independent_test_cutoff)

for.plot.ann <- for.plot %>%
  dplyr::mutate(sig = case_when(log10p > BF.frame$BF ~ "BF",
                                log10p > BF.frame$EIGEN ~ "EIGEN",
                                TRUE ~ "NONSIG"))

sig.colors <- c("red","#EE4266")
names(sig.colors) <- c("BF","EIGEN")

man.plot <-  ggplot2::ggplot() + 
  ggplot2::theme_bw() + 
  ggplot2::geom_point(data = for.plot.ann[which(for.plot.ann$sig != "NONSIG"),], 
                      mapping = aes(x = POS/1000000, 
                                    y = log10p,
                                    colour = sig),
                                    size=0.25) +
  ggplot2::scale_colour_manual(values = sig.colors) + 
  ggplot2::geom_point(data = for.plot[which(for.plot.ann$sig == "NONSIG"),], 
                      mapping = aes(x = POS/1000000, 
                                   y = log10p), 
                                   alpha = 0.5,
                                   size=0.25) +
  ggplot2::scale_y_continuous(expand = c(0,0), limits = c(0,BF + 6.7)) +
  scale_x_continuous(expand = c(0, 0), breaks = c(5, 10, 15, 20)) +
  ggplot2::geom_point(data = brig_genes, 
                      mapping=aes(x = POS/1000000, 
                                  y = log10p
                                  ),
                                  shape=25,
                                  size=2,
                                  fill="blue") +
  ggplot2::geom_hline(data = BF.frame, aes(yintercept = BF), linetype = 2) + 
  ggplot2::geom_hline(data = BF.frame, aes(yintercept = EIGEN), linetype = 3) + 
  ggplot2::labs(x = "",
                #y = expression(-log[10](italic(p)))) +
                y="") +
  ggplot2::theme(legend.position = "none", 
                 panel.grid = element_blank()) + 
  ggplot2::facet_grid(. ~ CHROM, scales = "free_x", space = "free") +
  ggplot2::theme(plot.title = element_text(face = "italic"))

brig_manplot <- man.plot
brig_manplot

```

3. c.t. Manhattan plot

```{r}
# in nextflow use sed to edit this field and make a copy of the .rmd for each trait
trait_name <- "telomere length"

#load gene list file for labelling
trop_genes <- read.delim("/projects/b1059/projects/Shriram/data/tropicalis/goi/trop_goi.tsv")

# load independent tests result
total_independent_tests <- read.table("/projects/b1059/projects/Shriram/data/tropicalis/nemascan_runs/Analysis_Results-20220209/Genotype_Matrix/total_independent_tests.txt", quote="\"", comment.char="", stringsAsFactors=FALSE)

independent_test_cutoff <- -log10(0.05/total_independent_tests[[1]])

# load processed mapping data. 
processed_mapping <- read.delim("/projects/b1059/projects/Shriram/data/tropicalis/nemascan_runs/Analysis_Results-20220209/Mapping/Processed/processed_length_AGGREGATE_mapping.tsv", stringsAsFactors=FALSE) %>%
  dplyr::mutate(CHROM = factor(CHROM, levels = c("I","II","III","IV","V","X","MtDNA"))) %>%
  dplyr::select(-marker) %>%
  tidyr::unite("marker", CHROM, POS, sep = ":", remove = F)


for.plot <- processed_mapping %>%
  dplyr::mutate(CHROM = as.factor(CHROM)) %>%
  dplyr::filter(CHROM != "MtDNA") %>%
  dplyr::mutate(algorithm = as.factor(algorithm))

BF <- processed_mapping %>% 
  dplyr::group_by(trait) %>% 
  dplyr::filter(log10p != 0) %>% 
  dplyr::mutate(BF = -log10(0.05/sum(log10p > 0, na.rm = T))) %>%
  dplyr::ungroup() %>%
  dplyr::select(BF) %>%
  unique(.) %>%
  as.numeric()


# ntests <- data.table::fread(tests) %>%
#  as.numeric()
# EIGEN <- -log10(0.05/ntests)
BF.frame <- processed_mapping %>%
  dplyr::select(trait) %>%
  dplyr::filter(!duplicated(trait)) %>%
  dplyr::mutate(BF = BF, EIGEN  = independent_test_cutoff)

for.plot.ann <- for.plot %>%
  dplyr::mutate(sig = case_when(log10p > BF.frame$BF ~ "BF",
                                log10p > BF.frame$EIGEN ~ "EIGEN",
                                TRUE ~ "NONSIG"))

sig.colors <- c("red","#EE4266")
names(sig.colors) <- c("BF","EIGEN")

man.plot <-  ggplot2::ggplot() + 
  ggplot2::theme_bw() + 
  ggplot2::geom_point(data = for.plot.ann[which(for.plot.ann$sig != "NONSIG"),], 
                      mapping = aes(x = POS/1000000, 
                                    y = log10p,
                                    colour = sig),
                                    size=0.25) +
  ggplot2::scale_colour_manual(values = sig.colors) + 
  ggplot2::geom_point(data = for.plot[which(for.plot.ann$sig == "NONSIG"),], 
                      mapping = aes(x = POS/1000000, 
                                   y = log10p), 
                                   alpha = 0.5,
                                   size=0.25) +
  ggplot2::scale_y_continuous(expand = c(0,0), limits = c(0,BF + 6.7)) +
  scale_x_continuous(expand = c(0, 0), breaks = c(5, 10, 15, 20)) +
  ggplot2::geom_point(data = trop_genes, 
                      mapping=aes(x = POS/1000000, 
                                  y = log10p
                                  ),
                                  shape=25,
                                  size=2,
                                  fill="blue") +
  ggplot2::geom_hline(data = BF.frame, aes(yintercept = BF), linetype = 2) + 
  ggplot2::geom_hline(data = BF.frame, aes(yintercept = EIGEN), linetype = 3) + 
  ggplot2::labs(x = "",
                y = expression(-log[10](italic(p)))) +
  ggplot2::theme(legend.position = "none", 
                 panel.grid = element_blank()) + 
  ggplot2::facet_grid(. ~ CHROM, scales = "free_x", space = "free") +
  ggplot2::theme(plot.title = element_text(face = "italic"))

trop_manplot <- man.plot
trop_manplot
```


Genotype Matrix: 


```{r}
brig_geno <- read_tsv('/projects/b1059/projects/Shriram/data/briggsae/nemascan_runs/Analysis_Results-20211219/Genotype_Matrix/Genotype_Matrix.tsv')
```


#Shortlisted genes: 

1. cbrmrt-1.2
https://wormbase.org/species/c_briggsae/gene/WBGene00030471#0-9fdg6-10

```{r}
#cbrmrt-1.2
cbrmrt1 <- brig_geno %>% filter(POS>=15329043 & POS<=15332022) %>% filter(CHROM=="I")
brig_temp1 <- cbrmrt1 %>% pivot_longer(
    cols=!CHROM & !POS & !REF & !ALT,
    names_to = "isotype",
    values_to = "altref"
)
brig_temp <- brig_temp1 %>% inner_join(cb_len_isotype) %>% filter(altref==-1) %>% dplyr::select(POS,length,isotype)
mrt_gene <- read_tsv('/projects/b1059/projects/Shriram/data/briggsae/goi/regional_gffs/cbrmrt1.tsv')

position_list <- brig_temp %>% count(POS) %>% dplyr::select(POS)

cbrmrt1_plot <- brig_temp %>% ggplot(aes(x=POS,y=length,group=POS)) + 
  geom_boxplot() + 
  geom_jitter(color="black", alpha=0.1) + 
  xlim(15329043,15332022) + 
  theme(axis.title = element_blank(),axis.text.y = element_blank(),axis.ticks.y = element_blank(), legend.position = "none") + 
  geom_vline(xintercept = c(mrt_gene$start),linetype="dashed",alpha=0.5) 

cbrmrt1_gene_plot <- ggplot(mrt_gene, aes(xmin = start, xmax = end, y = molecule, fill = gene)) + 
  gggenes::geom_gene_arrow(arrowhead_height = unit(3, "mm"), arrowhead_width = unit(1, "mm")) + 
  theme(axis.title = element_blank(),axis.text = element_blank(),axis.ticks = element_blank(), legend.position = "none") +
  geom_vline(xintercept = c(mrt_gene$start),linetype="dashed",alpha=0.5) 

#green intron, pink exon


p <- plot_grid(cbrmrt1_gene_plot,cbrmrt1_plot,ncol=1,rel_widths = c(1,1),rel_heights = c(1,6))
#p <- plot_grid(p,)

p

```

```{r}

iso_list <- brig_temp1 %>% filter(altref==-1) %>% dplyr::select(strain=isotype) %>% distinct() %>% add_column(status=1)
cbrmrt_strains <- cb_len %>% mutate(strain=fct_reorder(strain,length))

x <- as_tibble(setdiff(cbrmrt_strains$strain,iso_list$strain)) 
y <- x %>% add_column(status=-1)
y <- y %>% rename(strain=value)

z <- full_join(iso_list,y)
k <- inner_join(cb_len,z)

cbrmrt_bar <- k %>% mutate(strain=fct_reorder(strain,length)) %>%
      ggplot(aes(x=strain,y=length,fill=as.factor(status))) + 
        labs(fill="deletion") +
        geom_bar(stat='identity') +
        xlab("briggsae strains")

cbrmrt_bar
```




2. cbr-aex5
https://wormbase.org/species/c_briggsae/gene/WBGene00038888#0-9fdg6-10

```{r}
#cbr-aex-5
cbraex5 <- brig_geno %>% filter(POS>=14669282 & POS<=14679748) %>% filter(CHROM=="I")
brig_temp <- cbraex5 %>% pivot_longer(
    cols=!CHROM & !POS & !REF & !ALT,
    names_to = "isotype",
    values_to = "altref"
)
brig_temp <- brig_temp %>% inner_join(cb_len_isotype) %>% filter(altref==-1) %>% dplyr::select(POS,length) 
cbraex5_plot <- brig_temp %>% ggplot(aes(x=POS,y=length)) + geom_boxplot(alpha=0.2,aes(group=POS)) + geom_jitter(color="black", size=0.2, alpha=0.2) + xlim(14669282,14679748)
cbraex5_plot
```

